proposals/deliverable-proposals/profiles.md

```diff
@@ -12,11 +12,40 @@ and on resource constrained devices.
 Initially this work item will focus on delivering the Profile 1.1 specification.
 When TD 2.0 becomes available, appropriate clarifications and constraints will be delivered in a Profile 2.0 specification.
 
-In addition, specification work for other profiles, including the following profiles will be carried out:  
-* Digital Twin Profile (addressing behavioral, accessibility and UI requirements)
-* Cloud Profile (addressing requirements of large scale requirements)
-* Constraint Device Profile (addressing requirements of devices with a small footprint)
-* Profiles for other protocols (e.g. CoAP, MQTT)
+In addition, specification work for other profiles will be carried out. 
+Initial use cases and requirements for other profiles are described in https://www.w3.org/TR/wot-usecases .
+
+To address some of these use cases, the following profiles have been proposed:  
+
+### Digital Twin Profile
+ 
+Digital twins enable defining a model of a thing, person or process. They are used as digital proxies for counterparts in the real world to better understand situations in the real world and act on state changes. 
+
+Using a digital twin allows businesses to analyze their physical assets to troubleshoot in real time, predict future problems, minimize downtime, and perform simulations to create new business opportunities. A digital twin may also be called a twin or a shadow. 
+
+Digital twins have different roles. They could be models of individual discrete things, composite twins or organisational twins of more complex entities, such as factories or cities.
+Some use cases for digital twins have been described in the **WoT use cases and requirements** document, e.g. for Smart Cities, Production Monitoring, Health and Agriculture scenarios.
+
+The *Digital Twin Profile* enables using the thing description for defining digital twins that address these roles, use cases and requrements. 
+
+### Cloud Profile
+
+Large scale deployments, for example in production lines for industrial manufacturing consist of multiple machines, where each machine incorporates sensors for various values. A failure of a single machine can cause defective products or a stop of the entire production.
+
+Big data analysis enables to identify behavioral patterns across multiple production lines of the entire production plant and across multiple plants.
+
+The results of this analysis can be used for optimizing consumption of raw materials, checking the status of production lines and plants and predicting and preventing fault conditions.
+
+The *Cloud Profile* enables large scale deployments of devices across different manufacturers. 
```

Comment:
How would the Digital Twin Profile and Cloud Profile differ technically from the existing HTTP Baseline Profile, HTTP SSE Profile and HTTP Webhook Profile? These descriptions explain what digital twins and cloud deployments are, but they don't explain why they need their own profiles.

Bear in mind that profiles improve interoperability _within_ profiles, but not _between_ profiles. Defining a profile specific to digital twins could have the effect of making them non-interoperable with Web Things which aren't digital twins. Defining a profile specific to cloud deployments could have the effect of making them non-interoperable with Web Things not deployed in the "cloud". That would not be desirable.

I am actually currently implementing a cloud-based digital twin which uses the existing profiles and I don't see a need for profile specific to those use cases. However, I do see requirements experienced by cloud-based digital twins which could be applicable for a more horizontal profile.

For example, when creating a digital twin of a whole building it would be useful to be able to subscribe to multiple events and observe multiple properties of multiple Things over a single network socket to an on-premises building management system. This is not possible with the current profiles but is not necessarily specific to digital twins.

I think rather than the vertical sounding "digital twin profile" and "cloud profile" it might make more sense to create a "WebSocket Profile" (or extending the existing HTTP SSE Profile and HTTP Webhook Profile), which defines a protocol binding for subscribing to data from multiple Things at once. That would be an example of a tangible requirement for horizontal profiles.

Reply:
@benfrancis 
> How would the Digital Twin Profile and Cloud Profile differ technically from the existing HTTP Baseline Profile, HTTP SSE Profile and HTTP Webhook Profile? These descriptions explain what digital twins and cloud deployments are, but they don't explain why they need their own profiles.

These are use cases and scenarios that require additional analysis and requirements work. As a starting point the use cases and requirements document describes already some. I agree with your point that we should create horizontal profiles that address the needs of several stakeholders and provide OOTBI interoperability among their implementations.

> Bear in mind that profiles improve interoperability _within_ profiles, but not _between_ profiles. Defining a profile specific to digital twins could have the effect of making them non-interoperable with Web Things which aren't digital twins. Defining a profile specific to cloud deployments could have the effect of making them non-interoperable with Web Things not deployed in the "cloud". That would not be desirable.

This is not necessarily the case - If two profiles use the same baseline and just differ in the eventing protocol, the common baseline subset is still OOTBI. If these also put additional requirements and constraints on the baseline to satisfy accessibility requirements and enable an accessible UI, these will be compatible from a user's POV.

> I am actually currently implementing a cloud-based digital twin which uses the existing HTTP-based profiles and I don't see a need for profile specific to those use cases. However, I do see requirements for cloud-based digital twins which could be applicable for other more horizontal profiles.
> 
I agree, we should collect these requirements and define a common horizontal profile.

> For example, when creating a digital twin of a whole building it would be useful to be able to subscribe to multiple events and observe multiple properties of multiple Things over a single network socket to an on-premises building management system. This is not possible with the current profiles but is not necessarily specific to digital twins.
> 
This is an area, where composite and organisational twins have new requirements. 

> I think rather than the vertical sounding "digital twin profile" and deployment model specific "cloud profile" it might make more sense to create a "**WebSocket Profile**" (or an updated version of the existing HTTP SSE Profile or HTTP Webhook Profile), which defines a protocol binding for subscribing to data from multiple Things at once.
> 
I never considered the digital twin profile as a vertical profile, in my view it is part of many vertical application domains among which agriculture, buildings, energy, healthcare, manufacturing, smart cities are the obvious ones.

> That would be an example of a tangible requirement which cloud-based digital twins might have of other horizontal profiles, but I don't think profiles specific to digital twins or the cloud deployment model make sense.
Initially, I suggest we first discuss scenarios and collect the requirements before we discuss the protocols that may be part of a solution.

Reply:
@mlagally wrote:
> These are use cases and scenarios that require additional analysis and requirements work.

The current text reads as though these are new dedicated profiles which should be defined. Perhaps it would be better to say that the group should explore how new or existing profiles could fulfil the requirements of digital twins, cloud deployments and constrained devices, without jumping to the conclusion that these would all need to be separate profiles. We've discussed an example above of a requirement to observe multiple Things over a single connection - that could be fulfilled by a new profile, or an extension of an existing one.

> As a starting point the use cases and requirements document describes already some.

Those use cases are very high level and don't really answer my question. What I'm looking for are concrete technical requirements which come from these use cases that can not be fulfilled by the current profiles (or extensions of them), and rationale for why they require a dedicated profile (which by definition will not be fully interoperable with the existing profiles), as suggested here.

> If two profiles use the same baseline and just differ in the eventing protocol, the common baseline subset is still OOTBI.

Perhaps this could be justified if there was evidence that an event emitted by a digital twin or a cloud service is fundamentally different from an event which is emitted directly by a physical device. What I'm saying is that I've not seen any evidence yet that they are fundamentally different and that they therefore justify a separate (and therefore incompatible) event mechanism.

> I never considered the digital twin profile as a vertical profile, in my view it is part of many vertical application domains among which agriculture, buildings, energy, healthcare, manufacturing, smart cities are the obvious ones.

I agree that digital twins can span multiple vertical application domains. What I'd like to better understand is if and how a Web Thing representing a digital twin is different from any other Web Thing from a Consumer's point of view. Perhaps a digital twin is just one use case of a Web Thing, but a Consumer doesn't care whether that Web Thing represents a physical device or a simulation of one.

> Initially, I suggest we first discuss scenarios and collect the requirements before we discuss the protocols that may be part of a solution.

That makes sense.